balancer/weightedroundrobin/balancer.go

```diff
@@ -527,21 +592,35 @@ func (w *weightedSubConn) updateConnectivityState(cs connectivity.State) connect
 
 // weight returns the current effective weight of the subconn, taking into
 // account the parameters.  Returns 0 for blacked out or expired data, which
-// will cause the backend weight to be treated as the mean of the weights of
-// the other backends.
-func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration) float64 {
+// will cause the backend weight to be treated as the mean of the weights of the
+// other backends. If forScheduler is set to true, this function will emit
+// metrics through the mtrics registry.
+func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration, forScheduler bool) (weight float64) {
 	w.mu.Lock()
 	defer w.mu.Unlock()
+
+	weight = 0
+	if forScheduler {
+		defer endpointWeightsHandle.Record(w.metricsRecorder, weight, []string{w.target, w.locality}...)
```

Comment:
I don't think this works?  I think `weight` gets captured immediately.  You would need to wrap in a `func()` if you want to do it this way.

Reply:
Oh wait there actually is a correctness issue here since I don't set weight before returning for the case it passes the two conditionals that mean weight = 0 (blackout/weight expiration period).

Reply:
Ah haha yeah this is wrong wrapped with a func().